File: application/src/main/java/org/sheedon/common/handler/NotifyAppStateHandler.java
package org.sheedon.common.handler;

import android.app.Activity;

/**
 * 通知App状态更改处理者
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2021/7/26 6:20 下午
 */
public final class NotifyAppStateHandler {

    private static OnNotifyChangeListener listener;

    public static void setListener(OnNotifyChangeListener listener) {
        NotifyAppStateHandler.listener = listener;
    }

    public static void addActivity(Activity activity) {
        if (listener != null) {
            listener.addActivity(activity);
        }
    }

    public static void removeActivity(Activity activity) {
        if (listener != null) {
            listener.removeActivity(activity);
        }
    }

    public interface OnNotifyChangeListener {
        void addActivity(Activity activity);

        void removeActivity(Activity activity);
    }
}

File: application/src/main/java/org/sheedon/common/handler/NotifyAppStateHandler.kt
package org.sheedon.common.handler

import android.app.Activity

/**
 * 通知App状态更改处理者
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/1/5 9:38 下午
 */
class NotifyAppStateHandler private constructor() {

    var listener: OnNotifyChangeListener? = null

    companion object {
        private val instance = NotifyAppStateHandler()

        /**
         * 附加监听器
         * @param listener 通知App状态更改监听器
         */
        @JvmStatic
        fun attachListener(listener: OnNotifyChangeListener) {
            instance.listener = listener
        }

        /**
         * activity入栈
         * @param activity Activity
         */
        @JvmStatic
        fun pushActivity(activity:Activity){
            instance.listener?.pushActivity(activity)
        }

        /**
         * activity出栈
         * @param activity Activity
         */
        @JvmStatic
        fun popActivity(activity:Activity){
            instance.listener?.popActivity(activity)
        }

    }

    /**
     * 通知更改监听器
     */
    interface OnNotifyChangeListener {
        fun pushActivity(activity: Activity)

        fun popActivity(activity: Activity)
    }
}